rrfs-test/IODA/yaml/prepbufr_adpsfc.yaml

```diff
@@ -31,31 +31,36 @@ observations:
             query: "*/SID"
           stationElevation:
             query: "*/ELV"
+            type: float
```

Comment:
Just curious, if we don't specify "float", what's the default type? Integer?  @spanNOAA @delippi @HuiLiu-NOAA

Reply:
@guoqing-noaa, not specifying it is why it was int before. This is what it looks like in the bufr file:
```
010199  ELV                       2801.0  METER                         STATION ELEVATION
055007  TYP                        295.0  CODE TABLE                    PREPBUFR REPORT TYPE
```

Perhaps since the precision is only to the 1's digit then it assumes it is integer? For example TYP defaults to integer as well. That is my best guess.

Reply:
"TYP" also has one decimal even though it is 0. 

I checked a few prepbufr messages and it looks like the elevation does not have any meaningful decimals.
```
    #,  levels,ID,time,x,y,elv,DHR,type
  534   1  KVTXGPSS  20241004_1100  262.107   27.546    -3.0 -1.0000 153
  535   1  P802GPSS  20241004_1100  259.377   46.558   572.0 -1.0000 153
  536   1  P817GPSS  20241004_1100  281.489   40.146   384.0 -1.0000 153
  537   1  N515GPSS  20241004_1100  202.121   21.322    42.0 -1.0000 153
  538   1  N516GPSS  20241004_1100  245.969   51.150  1080.0 -1.0000 153
  539   1  N517GPSS  20241004_1100  280.344   43.713   142.0 -1.0000 153
  540   1  N518GPSS  20241004_1100  270.517   40.612   195.0 -1.0000 153
  541   1  N519GPSS  20241004_1100  262.351   35.467   360.0 -1.0000 153
  542   1  N520GPSS  20241004_1100  275.861   35.958   263.0 -1.0000 153
```
So we may also want to keep it as an integer.

@HuiLiu-NOAA Do you know what EDI wants for the elevation value? Integer or Float?

Reply:
It seems that an integer is good enough for station elevation height, from the ZTD bufr table:

9 Height of Station " 0   -400   15   Metres   to 1 m – geometric height relative to mean sea level

Reply:
Ah, I see. Thanks.

Reply:
@HuiLiu-NOAA Could you double check whether JEDI expects a float for station elevation? 
Donnie got an error when using integer elevation.

Reply:
my test with ZTD obs was fine with the integer elevation.

Reply:
<img width="762" alt="Screenshot 2024-10-04 at 10 20 34 AM" src="https://github.com/user-attachments/assets/38d23e6b-73ae-4a20-96a5-244ac64f5583">

Reply:
Actually, I think it was height that was giving me problems. I just also changed stationElevation thinking there would be a similar issue there. Can anyone confirm if height as an integer gives a problem? I can try to replicate the problem I was seeing with height

Is the preference to switch stationElevation back to integer?

Reply:
@delippi It is preferred to be consistent with the prepbufr information, i.e. using integer, unless it causes troubles in JEDI.

Reply:
seems no difference. even the float is set, the actual value is still integer.

Reply:
> seems no difference. even the float is set, the actual value is still integer.

@HuiLiu-NOAA, If you just checked it, then that is because I just changed both back to integer and reprocessed the data. I'm trying to replicate my error. I'm actually having trouble replicating it, so I might just change them back and if it comes up again we can easily make the change at that time.

Reply:
I looked at the original sfc obs input file in the OLD test case.